### ISSUE TITLE
Export to markdown

### ISSUE THREAD

**Issue by bshlgrs** [quip/quip-api#8]
Is there a way to export Quip documents to Markdown?

**Comment by caseymrm** [comment]
Currently we only return HTML. You may be able to get what you need using an HTML to Markdown library.

**Comment by jernejcic** [comment]
This is the one thing that keeps me from using Quip. I want to be able to have a backup of my Quip docs in MD files.

**Comment by mkeneqa** [comment]
You can export to Markdown now in 5.0.18

**Comment by hellosputnik** [comment]
If you can export to Markdown using Quip API (i.e., without an HTML to Markdown library), can you share an example, @mkeneqa?

I do not see that functionality in the documentation.

**Comment by mkeneqa** [comment]
@hellosputnik  - Not exactly sure what you mean here but the following screen shot is how one could go about exportint Quip notes to Markdown. Hope this helps.

![quipmarkdownexport](https://user-images.githubusercontent.com/3219890/30178229-752bc6b0-93c5-11e7-9010-bd586a7116d4.jpg)

**Comment by caseymrm** [comment]
@mkeneqa Quip has had that feature on web since 2015: https://quip.com/blog/markdown

This issue is about supporting it through the API. Might be a good task for somebody looking to get started on platform work?

**Comment by hellosputnik** [comment]
Ah, I knew it was too good to be true. I said using the [Quip API](https://quip.com/api/reference), not the client interface :)

**Comment by analyticalmonk** [comment]
Has there been any progress on this issue?
Specifically, can we export a quip document to markdown using the API?

**Comment by vjpr** [comment]
Just export as HTML and then convert HTML to markdown.

**Comment by analyticalmonk** [comment]
I wanted to know if the functionality had been included, just in case.
But thanks, html will do.

**Comment by aggrolite** [comment]
It's frustrating that we can export docs to Markdown via the UI, but *not* the API.

**Comment by sfbalg** [comment]
Yes, can we please make this happen?  Getting HTML is one thing, but then it gets mangled by whatever HTML-to-MD is popular this week and I have to spend the afternoon editing it by hand.

**Comment by coxley** [comment]
This is important to me, as well. To maintain formatting consistency between the document on Quip and outside, the UI export is super accurate. A lot of HTML -> Markdown conversions aren't for Quip's output HTML.

**Comment by amzn-xyxu** [comment]
+1, it would be super helpful if we can get Markdown Export via API.

**Comment by coxley** [comment]
For anyone following this issue, I wrote a library and CLI for converting Quip HTML to Markdown: https://github.com/coxley/quip2md

The list of supported features are in the README :)

**Comment by CyTheGuy** [comment]
+1 

I was wonder if there is any traction with this. I was told in June 2018 that it was in the works so hoping there is some light at the end of the tunnel.

**Comment by parrachia** [comment]
o/

**Comment by jboursiquot** [comment]
Exporting to Markdown via the API would be quite useful. Alas, one has to resort to exporting to HTML  and then try to convert.

I'm wondering if the lack of this feature is due in part to the complexities involved in converting the rich elements that can appear in a Quip thread into Markdown. Converting things like Process Bars, Live Apps, Polls, etc to a Markdown-equivalent might be prohibitive. Perhaps such documents could simply be treated as "Incompatible" by having the API respond with an HTTP `4XX` status code.

In any case, it would be good to have an official recommendation for export treatment from the Quip team here. Thanks.

**Comment by sonnenkern** [comment]
> This is the one thing that keeps me from using Quip. I want to be able to have a backup of my Quip docs in MD files.

I think HTML is more convenient, because you can have more beautiful ans more standard output with images and references. You can try my tool, I've developed for Quip backup purpose:
www.quip-export.com (online web-version)
https://www.npmjs.com/package/quip-export (as npm-module and command line tool)

**Comment by sonnenkern** [comment]
> Has there been any progress on this issue?
> Specifically, can we export a quip document to markdown using the API?

You can try my tool, I've developed it special for Quip backup purpose:
www.quip-export.com (online web-version)
https://www.npmjs.com/package/quip-export (as npm-module and command line tool)

**Comment by Phando** [comment]
I see export->markdown in the menu, but when I click it nothing happens.

**Comment by xavierfeltin** [comment]
Hi,
I started to use the Quip API. The markdown export is still not present in the Quip API.
Is something planned to support it ?
Thanks.

**Comment by KoStard** [comment]
Interested in this feature as well.
